extensions/wikia/CategoryPage3/CategoryPage3Hooks.class.php

```diff
@@ -54,4 +66,39 @@ static public function onArticleFromTitle( $title, &$article ): bool {
 
 		return true;
 	}
+
+	/**
+	 * Redirect to the canonical category URL if any of the older
+	 * pagination URL params were used.
+	 *
+	 * @param Title $title
+	 * @param $unused
+	 * @param OutputPage $output
+	 * @param User $user
+	 * @param WebRequest $request
+	 * @param MediaWiki $mediawiki
+	 * @return bool
+	 */
+	static public function onBeforeInitialize(
+		Title $title, $unused, OutputPage $output,
+		User $user, WebRequest $request, MediaWiki $mediawiki
+	): bool {
+		if ( !$title->inNamespace( NS_CATEGORY ) ||
+			$user->isLoggedIn()
```

Comment:
I think we should check the current layout in addition to that. There is no existing method that could be used here, so I'll update it as a part of the layout selector task.